tutorials/persistance-comparison-series/File/file.md

```diff
@@ -0,0 +1,345 @@
+
+```cs
+private void Update() // 1
+{
+    // Check if a key was pressed.
+    if (Input.GetKey(KeyCode.LeftShift)) // 2
+    {
+        // Set the LeftShift key.
+        keyPressed = KeyCode.LeftShift; // 3
+    }
+    else if (Input.GetKey(KeyCode.LeftControl)) // 2
+    {
+        // Set the LeftControl key.
+        keyPressed = KeyCode.LeftControl; // 3
+    }
+    else // 4
+    {
+        // In any other case reset to default and consider it unmodified.
+        keyPressed = default; // 5
+    }
+}
+```
+
+Now to saving the data when a click happens:
+
+```cs
+private void OnMouseDown() // 6
+{
+    // Check if a key was pressed.
+    if (keyPressed == KeyCode.LeftShift) // 7
+    {
+        // Increment the Shift hit count.
+        hitCountShift++; // 8
+    }
+    else if (keyPressed == KeyCode.LeftControl) // 7
+    {
+        // Increment the Control hit count.
+        hitCountControl++; // 8
+    }
+    else // 9
+    {
+        // If neither Shift nor Control was held, we increment the unmodified hit count.
+        hitCountUnmodified++; // 10
+    }
+
+    // 11
+    // Create a string array with the three hit counts.
+    string[] stringArray = {
+        hitCountUnmodified.ToString(),
+        hitCountShift.ToString(),
+        hitCountControl.ToString()
+    };
+
+    // 12
+    // Save the entries, line by line.
+    File.WriteAllLines(HitCountFileUnmodified, stringArray);
+}
+```
+
+Whenever a mouse click is detected on the capsule (6) we can then perform a similar check to what happened in `Update()` only that we use `keyPressed` instead of `Input.GetKey()` here.
+
+Check if `keyPressed` was set to `KeyCode.LeftShift` or `KeyCode.LeftControl` (7) and if so, increment the corresponding hit count (8). If no modifier was used (9), increment the `hitCountUnmodified`.
+
+As seen in the last section we need to create a string that can be saved in the file. There is a second function on `File` that accepts a a string array and then saves each entry in one line: `WriteAllLines()`.
+
+Knowing this we create an array containing the three hit counts (11) and pass this one on to `File.WriteAllLines()`.
+
+Start the game, click the capsule using Shift and Control. You should see the three counters in the Inspector.
+
+<img src="images/07_hit_count_extended_editor.jpg" alt="File hitCountExtended.txt" width=""/>
+
+After stopping the game and therefore saving the data a new file `hitCountFileExtended.txt` should exist in your project folder. Have look at it, it should look something like this:
+
+<img src="images/08_hit_count_extended_file.jpg" alt="File hitCountExtended.txt" width=""/>
+
+As long as all the data saved to those lines belongs together, the file can be one option. If you have several different properties you might create multiple files. Alternatively you can save all the data into the same file using a bit of structure. One of the possible approaches to structuring your data will be shown in the next section.
```

Comment:
We should mention that the downside of this approach is that the numbers are not associated with the actual properties, which means that if someone refactors the code and reorders the checks, they'd be reading incorrect values.

Reply:
a8c02d9